core/math/src/main/java/org/arakhne/afc/math/tree/node/AbstractParentlessTreeNode.java

```diff
@@ -247,7 +248,7 @@ public String toString() {
 	@Override
 	public D[] getAllUserData(D[] array) {
 		if (this.data == null) {
-			return null;
+			return  (D[]) Array.newInstance(array.getClass(),0);
```

Comment:
- Would you like me to create a single PR for squid:S00122 issue ; which
   is about more than one statement being on the same line. There is a 2h TD
   for this issue in the project.


On Wed, Jun 8, 2016 at 7:50 PM, Stéphane Galland <notifications@github.com>
wrote:

> In
> core/math/src/main/java/org/arakhne/afc/math/tree/node/AbstractParentlessTreeNode.java
> <https://github.com/gallandarakhneorg/afc/pull/109#discussion_r66293623>:
>
> > @@ -247,7 +248,7 @@ public String toString() {
> >  	@Override
> >  	public D[] getAllUserData(D[] array) {
> >  		if (this.data == null) {
> > -			return null;
> > +			return  (D[]) Array.newInstance(array.getClass(),0);
>
> Caution: array parameter could be null.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/gallandarakhneorg/afc/pull/109/files/1421272924d9296cda07efbd5a2a3cb0295f1a28#r66293623>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AR09BIs4La5KvhXQP-wf9tK2LJwhhLF5ks5qJvLSgaJpZM4IxE0I>
> .
>